source/common/http/conn_manager_impl.cc

```diff
@@ -1058,6 +1079,16 @@ void ConnectionManagerImpl::ActiveStream::encodeHeaders(ActiveStreamEncoderFilte
   }
 }
 
+void ConnectionManagerImpl::ActiveStream::addEncodedTrailers(HeaderMapPtr&& trailers) {
+  if (state_.filter_call_state_ & FilterCallState::LastDataFrame) {
+    response_trailers_ = std::move(trailers);
+    state_.local_complete_ = false;
```

Comment:
this was necessary because `state_.local_complete_` gets set to true at the *start* of `encodeData`, so we have to undo that so that the ASSERT in `commonEncodePrefix` doesn't fire at the start of `encodeTrailers`